device/device.yaml

```diff
@@ -8,31 +8,22 @@ components:
   schemas:
     Device.Device:
       description: >-
-        An abstract container for emulated devices.
+        An abstract container for emulated protocol devices.
+        This can grow to accomodate other top level emulated protocols such 
+        as ipv4Loopback, ipv6Loopback etc.
       type: object
-      required: [name]
+      required: [name, choice]
       properties:
         name: '../common/common.yaml#/components/x-inline/UniqueName'
-        devices_per_port: 
+        port_name: 
           description: >-
-            The number of emulated devices that will be created on each port.
-          type: integer
-          default: 1
-        ethernets:
+            The name of a port that will host the emulated device stack.
+          type: string
+          x-constraint: '#/components/schemas/Port.Port.properties.name'
+        choice:
           description: >-
-            An unordered list of 0..n emulated ethernet protocols.
-          type: array
-          items:
-            $ref: './ethernet.yaml#/components/schemas/Device.Ethernet'
-        devices:
-          description: >-
-            An unordered list of 0..n emulated device containers.
-          type: array
-          items:
-            $ref: '#/components/schemas/Device.Device'        
-        networks:
-          description: >-
-            An unordered list of 0..n emulated device containers.
-          type: array
-          items:
-            $ref: './network.yaml#/components/schemas/Device.Network'        
+            The type of top level emulated protocol device. 
+          type: string
+          enum: [ethernet]
```

Comment:
it's more than ethernet for the enum?  or are those basic/initial set of choices coming in another PR?